### PR TITLE
Enable inactivity check on ovndb connection

### PR DESCRIPTION
**Pull request by fanriming**
# Pull Request

- [ ] Make sure you have followed [Kube-OVN Code Style](https://github.com/kubeovn/kube-ovn/blob/master/CODE_STYLE.md).

## What type of this PR

Examples of user facing changes:

- Features
- Bug fixes

Optimize the reconnection logic of the ovnnb client to avoid process suspension issues caused by frequent reconnection initialization in large-scale clusters.
 The following are reference links for ovn-kubernetes and libovsdb:
https://github.com/ovn-org/ovn-kubernetes/blob/52e4cadc81bcbc779b85bb1831932b442fbeeeb0/go-controller/pkg/libovsdb/libovsdb.go#L84
https://github.com/ovn-org/libovsdb/pull/280/files

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -270,7 +270,6 @@ require (
 replace (
 	github.com/mdlayher/arp => github.com/kubeovn/arp v0.0.0-20240218024213-d9612a263f68
 	github.com/openshift/client-go => github.com/openshift/client-go v0.0.1
-	github.com/ovn-org/libovsdb => github.com/kubeovn/libovsdb v0.0.0-20240218023647-f0bc3ce57fcd
```
Comment (oilbeater) [inline go.mod:273]: We replaced the libovsdb with a forked one that already contains `WithInactivityCheck`

**Review by oilbeater — CHANGES_REQUESTED** [review]

File: pkg/ovsdb/client/client.go [hunk 2]
```diff
@@ -25,7 +25,9 @@ const (
 	ICNBDB = "icnbdb"
 	ICSBDB = "icsbdb"
 )
-const timeout = 3 * time.Second
+
+const connectTimeout time.Duration = time.Second * 20
+const inactivityTimeout time.Duration = time.Second * 180
```
Comment (zhangzujian) [inline pkg/ovsdb/client/client.go:30]:
```suggestion
const (
	connectTimeout    time.Duration = time.Second * 20
	inactivityTimeout time.Duration = time.Second * 180
)
```

Comment (zhangzujian) [inline pkg/ovsdb/client/client.go:30]: I think we should make the timeout configurable.

File: go.mod (hunk 1 above)
Comment (fanriming) [reply go.mod]: fix

File: pkg/ovsdb/client/client.go (hunk 2 above)
Comment (fanriming) [reply pkg/ovsdb/client/client.go]: done

**Comment by oilbeater** [comment]
@fanriming the conflict should be resolved.

File: pkg/controller/config.go [hunk 3]
```diff
@@ -106,12 +108,14 @@ type Configuration struct {
...
-		argOvnSbAddr            = pflag.String("ovn-sb-addr", "", "ovn-sb address")
-		argOvnTimeout           = pflag.Int("ovn-timeout", 60, "The seconds to wait ovn command timeout")
-		argCustCrdRetryMinDelay = pflag.Int("cust-crd-retry-min-delay", 1, "The min delay seconds between custom crd two retries")
-		argCustCrdRetryMaxDelay = pflag.Int("cust-crd-retry-max-delay", 20, "The max delay seconds between custom crd two retries")
-		argKubeConfigFile       = pflag.String("kubeconfig", "", "Path to kubeconfig file with authorization and master location information. If not set use the inCluster token.")
+		argOvnNbAddr              = pflag.String("ovn-nb-addr", "", "ovn-nb address")
+		argOvnSbAddr              = pflag.String("ovn-sb-addr", "", "ovn-sb address")
+		argOvnTimeout             = pflag.Int("ovn-timeout", 60, "The seconds to wait ovn command timeout")
+    argOvsDbConTimeout        = pflag.Int("ovsdb-con-timeout", 20, "")
+    argOvsDbInactivityTimeout = pflag.Int("ovsdb-inactivity-timeout", 180, "")
```
Comment (zhangzujian) [inline pkg/controller/config.go:115]:
```suggestion
		argOvnTimeout             = pflag.Int("ovn-timeout", 60, "The seconds to wait ovn command timeout")
		argOvsDbConTimeout        = pflag.Int("ovsdb-con-timeout", 20, "")
		argOvsDbInactivityTimeout = pflag.Int("ovsdb-inactivity-timeout", 180, "")
```

File: pkg/controller/config.go [hunk 4]
```diff
@@ -106,12 +108,14 @@ type Configuration struct {
...
-		argOvnSbAddr            = pflag.String("ovn-sb-addr", "", "ovn-sb address")
-		argOvnTimeout           = pflag.Int("ovn-timeout", 60, "The seconds to wait ovn command timeout")
-		argCustCrdRetryMinDelay = pflag.Int("cust-crd-retry-min-delay", 1, "The min delay seconds between custom crd two retries")
-		argCustCrdRetryMaxDelay = pflag.Int("cust-crd-retry-max-delay", 20, "The max delay seconds between custom crd two retries")
-		argKubeConfigFile       = pflag.String("kubeconfig", "", "Path to kubeconfig file with authorization and master location information. If not set use the inCluster token.")
+		argOvnNbAddr              = pflag.String("ovn-nb-addr", "", "ovn-nb address")
+		argOvnSbAddr              = pflag.String("ovn-sb-addr", "", "ovn-sb address")
+		argOvnTimeout             = pflag.Int("ovn-timeout", 60, "The seconds to wait ovn command timeout")
+		argOvsDbConTimeout        = pflag.Int("ovsdb-con-timeout", 20, "")
+		argOvsDbInactivityTimeout = pflag.Int("ovsdb-inactivity-timeout", 180, "")
```
Comment (zhangzujian) [inline pkg/controller/config.go:115]: The default timeout values should be less for small clusters.

Comment (fanriming) [reply pkg/controller/config.go:115]: I refer to ovn-kubernetes for this value, what default value do you recommend?

File: pkg/ovs/ovn.go [hunk 5]
```diff
@@ -53,7 +53,7 @@ func NewLegacyClient(timeout int) *LegacyClient {
 	}
 }
 
-func NewOvnNbClient(ovnNbAddr string, ovnNbTimeout int) (*OVNNbClient, error) {
+func NewOvnNbClient(ovnNbAddr string, ovnNbTimeout int, ovsDbConTimeout int, ovsDbInactivityTimeout int) (*OVNNbClient, error) {
```
Comment (zhangzujian) [inline pkg/ovs/ovn.go:56]:
```suggestion
func NewOvnNbClient(ovnNbAddr string, ovnNbTimeout, ovsDbConTimeout, ovsDbInactivityTimeout int) (*OVNNbClient, error) {
```

File: pkg/ovs/ovn.go [hunk 6]
```diff
@@ -105,7 +112,7 @@ func NewOvnNbClient(ovnNbAddr string, ovnNbTimeout int) (*OVNNbClient, error) {
 	return c, nil
 }
 
-func NewOvnSbClient(ovnSbAddr string, ovnSbTimeout int) (*OVNSbClient, error) {
+func NewOvnSbClient(ovnSbAddr string, ovnSbTimeout int, ovsDbConTimeout int, ovsDbInactivityTimeout int) (*OVNSbClient, error) {
```
Comment (zhangzujian) [inline pkg/ovs/ovn.go:115]:
```suggestion
func NewOvnSbClient(ovnSbAddr string, ovnSbTimeout, ovsDbConTimeout, ovsDbInactivityTimeout int) (*OVNSbClient, error) {
```

**Review by zhangzujian — CHANGES_REQUESTED** [review]

**Review by zhangzujian — APPROVED** [review]

**Comment by oilbeater** [comment]
@fanriming please fix the DCO

**Pull request merged**